src/main/java/gift/dto/ProductCreateRequest.java

```diff
@@ -0,0 +1,34 @@
+package gift.dto;
+
+import gift.entity.Product;
+
+public class ProductCreateRequest {
+    private String name;
+    private Long price;
+    private String imageUrl;
+
+    public ProductCreateRequest() { }
+
+    public void setName(String name) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("상품 이름은 필수입니다.");
+        }
+        this.name = name;
+    }
+    public void setPrice(Long price) {
+        if (price == null || price <= 0) {
+            throw new IllegalArgumentException("상품 가격은 0보다 커야 합니다.");
+        }
+        this.price = price;
+    }
+    public void setImageUrl(String imageUrl) {
+        if (imageUrl == null || imageUrl.isEmpty()) {
+            throw new IllegalArgumentException("상품 이미지 URL은 필수입니다.");
+        }
+        this.imageUrl = imageUrl;
+    }
+
+    public Product toProduct() {
+        return new Product(null, name, price, imageUrl);
+    }
+}
```

Comment:
record 키워드로 ProductCreateRequest를 선언하면, 내부 필드가 자동으로 final로 설정되어 불변(immutable) 객체가 됩니다.
그렇기 때문에 setter 메서드를 제거하면, 객체 생성 이후에는 값이 변경될 일이 없어져 더 안전하게 객체를 관리할 수 있어요 😄 
불변 객체는 한 번 만들어진 이후 상태가 변하지 않기 때문에, 예기치 않은 값 변경을 막고 버그 가능성을 줄이는 데 큰 도움이 됩니다!
불변 객체로 만들어지면 변경 될일이 없으니까 안전하겠죠?
질문에 record 키워드를 사용하면 어떻게 검증을 하는지를 여쭤봐주셨는데, 저라면 다음과 같이 ProductCreateRequest를 생성할 때 한번만 검증을 하도록 하고 setter 메소드를 지워서 객체의 필드 값의 변경을 막을 것 같아요 !

```suggestion
public record ProductCreateRequest(String name, Long price, String imageUrl) {

    public ProductCreateRequest {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("상품 이름은 필수입니다.");
        }
        if (price == null || price <= 0) {
            throw new IllegalArgumentException("상품 가격은 0보다 커야 합니다.");
        }
        if (imageUrl == null || imageUrl.isBlank()) {
            throw new IllegalArgumentException("상품 이미지 URL은 필수입니다.");
        }
    }

    public Product toProduct() {
        return new Product(null, name, price, imageUrl);
    }
}
```
+ setter 메소드가 왜 안좋은지에 대해서 고민해보시고, 불변 객체가 왜 좋은지에 대해서도 같이 고민하면 좋을 것 같아요.